### PR TITLE
PR: Prevent segfaults when importing QtAwesome out of a of QApplication

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes spyder-ide/spyder#4014
Fixes spyder-ide/spyder#4049

### CODE REVIEW

File: qtawesome/iconic_font.py [hunk 1]
```diff
@@ -247,17 +251,18 @@ def icon(self, *names, **kwargs):
...
+                                                          general_options,
+                                                          names[i]))
 
-        # Process high level API
-        api_options = parsed_options
+            # Process high level API
+            api_options = parsed_options
 
-        return self._icon_by_painter(self.painter, api_options)
+            return self._icon_by_painter(self.painter, api_options)
```
Comment (goanpeca) [inline qtawesome/iconic_font.py]: Maybe the else should return an empty QIcon plus some [warning](https://docs.python.org/2/library/warnings.html#warnings.warn)?

```
else:
   warning.warn('You need to have a running QApplication to use QtAwesome!')
   return QIcon()
```

**Comment by goanpeca** [comment]
@dalthviz could we create a test that does the direct import (Like what @jitseniesen pointed out)

Just create a new tests folder inside the main module, ok?

File: qtawesome/tests/test_qtawesome.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Tests for QtAwesome
+"""
+# Standard library imports
+import subprocess
+
+# Test Library imports
+import pytest
+
+def test_segfault_import():
+    subprocess.call('python -c qtawesome')
```
Comment (ccordoba12) [inline qtawesome/tests/test_qtawesome.py]: Please this to

    output_number = subprocess.call('python -c qtawesome')
    assert  output_number == 0

Remember that you always have to `assert` something on tests :-)

Comment (goanpeca) [inline qtawesome/tests/test_qtawesome.py]: Should we try to import every file in the submodule also?

File: qtawesome/iconic_font.py (hunk 1 above)
Comment (ccordoba12) [inline qtawesome/iconic_font.py]: I think adding the warning is fine ;-)

File: qtawesome/tests/test_qtawesome.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+r"""
+Tests for QtAwesome
```
Comment (goanpeca) [inline qtawesome/tests/test_qtawesome.py]: dot at the end of a docstring (it is a sentence)

Comment (ccordoba12) [inline qtawesome/tests/test_qtawesome.py]: We don't have to be too strict about style on test files, but sure, why not :-)

File: qtawesome/tests/test_qtawesome.py (hunk 2 above)
Comment (ccordoba12) [inline qtawesome/tests/test_qtawesome.py]: The most important import's for QtAwesome are on its `__init__.py`, so I don't see the need for that.

**Comment by ccordoba12** [comment]
@ghisvail, this is kind of important in case you want to pick it up for Debian.

**Comment by ccordoba12** [comment]
@dalthviz, please also add the new tests folder to `MANIFEST.in`, like we do it in `qtpy`:

https://github.com/spyder-ide/qtpy/blob/master/MANIFEST.in#L5

That's necessary to add our tests to our released packages too, which was requested by @ghisvail for Debian.

**Comment by goanpeca** [comment]
Then can we also update the CI code to run the test?

https://github.com/spyder-ide/qtawesome/blob/master/appveyor.yml#L60
`- "%CMD_IN_ENV% py.test qtawesome"`

https://github.com/spyder-ide/qtawesome/blob/master/circle.yml
```
    - export PATH="$HOME/miniconda/bin:$PATH" && source activate test && py.test qtawesome: # note the colon
        parallel: true
```

**Comment by ccordoba12** [comment]
Ok, sure. All of that have to be part of this PR.

@dalthviz, sorry for putting you more work ;-)

**Comment by ghisvail** [comment]
I can cherry-pick this, but have to wait for the last upload of qtawesome-0.4.3 to transition with the rest of the stack, i.e. qtpy-1.2.1 and spyder-3.1.2. Otherwise, none of them will make it for the freeze.

**Comment by ccordoba12** [comment]
@dalthviz, I know why our tests are failing. I'll help you to fix them :-)

**Comment by ccordoba12** [comment]
Ok, I think this one is ready! Thanks @dalthviz for your hard work to solve this nasty problem :-)

**Pull request merged**